tests/cases/compiler/contextualTypeShouldBeLiteral.ts

```diff
@@ -0,0 +1,53 @@
+interface X {
+    type: 'x';
+    value: string;

```

Comment:
can you add a method that makes `this` of the object literal show up in the type baselines? I want to make sure it's not just `value: 'done'`, but `value: 'none' | 'done'` for the `type: 'y'` case.